sql/src/main/java/io/crate/planner/statement/CopyFromPlan.java

```diff
@@ -0,0 +1,420 @@
+    }
+
+    public AnalyzedCopyFrom copyFrom() {
+        return copyFrom;
+    }
+
+    @Override
+    public StatementType type() {
+        return StatementType.COPY;
+    }
+
+    @Override
+    public void executeOrFail(DependencyCarrier dependencies,
+                              PlannerContext plannerContext,
+                              RowConsumer consumer,
+                              Row params,
+                              SubQueryResults subQueryResults) {
+        var boundedCopyFrom = createStatement(
+            copyFrom,
+            plannerContext.transactionContext(),
+            plannerContext.functions(),
+            params, subQueryResults);
+
+        ExecutionPlan plan = planCopyFromExecution(
+            dependencies.clusterService().state().nodes(),
+            boundedCopyFrom, plannerContext);
+        NodeOperationTree nodeOpTree = NodeOperationTreeGenerator
+            .fromPlan(plan, dependencies.localNodeId());
+
+        dependencies.phasesTaskFactory()
+            .create(plannerContext.jobId(), List.of(nodeOpTree))
+            .execute(consumer, plannerContext.transactionContext());
+    }
+
+    @VisibleForTesting
+    public static BoundedCopyFrom createStatement(AnalyzedCopyFrom copyFrom,
+                                                  CoordinatorTxnCtx txnCtx,
+                                                  Functions functions,
+                                                  Row parameters,
+                                                  SubQueryResults subQueryResults) {
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
+            txnCtx,
+            functions,
+            x,
+            parameters,
+            subQueryResults
+        );
+
+        String partitionIdent;
+        if (!copyFrom.table().partitionProperties().isEmpty()) {
+            partitionIdent = PartitionPropertiesAnalyzer
+                .toPartitionName(
+                    copyFrom.tableInfo(),
+                    Lists2.map(copyFrom.table().partitionProperties(), x -> x.map(eval)))
+                .ident();
+        } else {
+            partitionIdent = null;
+        }
+        var properties = copyFrom.properties().map(eval);
+        var nodeFiltersPredicate = discoveryNodePredicate(
+            properties.properties().getOrDefault(NodeFilters.NAME, null));
+        var settings = genericPropertiesToSettings(properties);
+        var inputFormat = settingAsEnum(
+            FileUriCollectPhase.InputFormat.class,
+            settings.get(INPUT_FORMAT_SETTING.getKey(), INPUT_FORMAT_SETTING.getDefault(Settings.EMPTY)));
+        // TODO make FileUriCollectPhase ctor accept an uri of the List<String>
+        // instead of the Symbol type, such as the uri can be evaluated and converted
+        // to the required type already at this stage, but not later on in FileCollectSource.
+        var boundedURI = validateAndConvertToLiteral(eval.apply(copyFrom.uri()));
```

Comment:
I was thinking of changing the `FileUriCollectPhase` ctor to accept the URI of the `List<String>` type, but it'd require a small serialization change. I'd a short chat with jordi, and he was in favour of following up on it separately and keep the `FileUriCollectPhase` and `FileCollectSource` in the context of this change. Therefore, the uri symbol is evaluated and then converted back to literal.